packages/next-on-pages/templates/_worker.js/utils/images.ts

```diff
@@ -0,0 +1,184 @@
+ * @returns URL to the Cloudflare CDN image resizing endpoint.
+ */
+export function buildCdnCgiImageUrl(
+	requestUrl: URL,
+	imageUrl: URL,
+	{ width, quality, format }: RequestInitCfPropertiesImage
+): string {
+	const opts = [];
+
+	if (width) opts.push(`width=${width}`);
+	if (quality) opts.push(`quality=${quality}`);
+	if (format) opts.push(`format=${format}`);
+
+	const imageHref =
+		requestUrl.origin === imageUrl.origin
+			? imageUrl.pathname.slice(1)
+			: imageUrl.href;
+
+	return `${requestUrl.origin}/cdn-cgi/image/${opts.join(',')}/${imageHref}`;
+}
+
+/**
+ * Formats the given response to match the images configuration spec from the Vercel build output
+ * config.
+ *
+ * Applies headers for `Content-Security-Policy` and `Content-Disposition`, if defined in the config.
+ *
+ * https://vercel.com/docs/build-output-api/v3/configuration#images
+ *
+ * @param resp Response to format.
+ * @param imageUrl Image URL that was resized.
+ * @param config Images configuration from the Vercel build output.
+ * @returns Formatted response.
+ */
+export function formatResp(
+	resp: Response,
+	imageUrl: URL,
+	config?: VercelImagesConfig
+): Response {
+	const newHeaders = new Headers(resp.headers);
+
+	if (config?.contentSecurityPolicy) {
+		newHeaders.set('Content-Security-Policy', config.contentSecurityPolicy);
+	}
+
+	if (config?.contentDispositionType) {
+		const fileName = imageUrl.pathname.split('/').pop();
+		const contentDisposition = fileName
+			? `${config.contentDispositionType}; filename="${fileName}"`
+			: config.contentDispositionType;
+
+		newHeaders.set('Content-Disposition', contentDisposition);
+	}
+
+	return new Response(resp.body, { headers: newHeaders });
+}
+
+/**
+ * Handles image resizing requests.
+ *
+ * @param request Incoming request.
+ * @param requestUrl Incoming request's URL.
+ * @param config Images configuration from the Vercel build output.
+ * @returns Resized image response if the request is valid, otherwise a 400 response.
+ */
+export async function imageResizing(
+	request: Request,
+	requestUrl: URL,
+	config?: VercelImagesConfig
+): Promise<Response> {
+	const opts = getResizingProperties(request, requestUrl, config);
+
+	if (!opts) {
+		return new Response('Invalid image resizing request', { status: 400 });
+	}
+
+	const { imageUrl } = opts;
+
+	// NOTE: Pages does not support the RequestInit image resizing yet.
+	// const imageReq = new Request(imageUrl, { headers: request.headers });
+	// const imageResp = await fetch(imageReq, { cf: { image: options } });
+	// if (imageResp.status === 200) return formatResp(imageResp, imageUrl, config);
+
+	// NOTE: Pages also doesn't seem to support calling the `/cdn-cgi/image` endpoint either.
+	// const cdnCgiImageUrl = buildCdnCgiImageUrl(requestUrl, imageUrl, options);
+	// const cdnCgiResp = await fetch(cdnCgiImageUrl, { headers: request.headers });
+	// if (cdnCgiResp.status === 200) return formatResp(cdnCgiResp, imageUrl, config);
```

Comment:
I had a chat with @GregBrimble and if I understood correctly it doesn't seem like this is going to be supported in the future

so I think maybe we should drop this entirely (alongside `buildCdnCgiImageUrl`) as it is never going to be applied anyways

Greg?

Reply:
I've gone ahead and removed it for now. tbh, I wouldn't really be a fan of falling back to that endpoint even if it were possible.